### PR TITLE
Authorize secrets

### LINKED ISSUE #11283: Authorize Secret Service operations.
Part of https://github.com/influxdata/influxdb/issues/10814

### PR DESCRIPTION
**Pull request by leodido**
Closes #11283 

_Briefly describe your proposed changes:_

This PR introduces a secrets service for authorizing requests.

  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] http/swagger.yml updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: authorizer/secret.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+// LoadSecret checks to see if the authorizer on context has read access to the secret key provided.
+func (s *SecretService) LoadSecret(ctx context.Context, orgID influxdb.ID, key string) (string, error) {
+	secret, err := s.s.LoadSecret(ctx, orgID, key)
+	if err != nil {
+		return "", err
+	}
+
+	if err := authorizeReadSecret(ctx, orgID); err != nil {
```
Comment (desa) [inline authorizer/secret.go]: Can this come before the call to `s.s.LoadSecret` since we already have the orgID?

File: authorizer/secret.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+// GetSecretKeys checks to see if the authorizer on context has read access to all the secrets belonging to orgID.
+func (s *SecretService) GetSecretKeys(ctx context.Context, orgID influxdb.ID) ([]string, error) {
+	secrets, err := s.s.GetSecretKeys(ctx, orgID)
+	if err != nil {
+		return []string{}, err
+	}
+
+	if err := authorizeReadSecret(ctx, orgID); err != nil {
```
Comment (desa) [inline authorizer/secret.go]: ditto previous comment.

File: authorizer/secret.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+		return err
+	}
+
+	return nil
+}
+
+// PutSecrets checks to see if the authorizer on context has read and write access to the secret keys provided.
+func (s *SecretService) PutSecrets(ctx context.Context, orgID influxdb.ID, m map[string]string) error {
+	// PutSecrets operates on intersection betwen m and keys beloging to orgID.
+	// We need to have read access to those secrets since it deletes the secrets (within the intersection) that have not be overridden.
```
Comment (desa) [inline authorizer/secret.go]: 👍 seems reasonable that you should have both permissions to do this.

File: authz.go [hunk 4]
```diff
@@ -115,6 +115,8 @@ const (
 	MacrosResourceType = ResourceType("macros") // 8
 	// ScraperResourceType gives permission to one or more scrapers.
 	ScraperResourceType = ResourceType("scrapers") // 9
+	// SecretsResourceType gives permission to one or more scrapers.
```
Comment (desa) [inline authz.go]: `s/scrapers/secrets/`

File: authorizer/secret.go (hunk 3 above)
Comment (leodido) [reply authorizer/secret.go]: I tried to explain the reason behind this choice in the comments for future reference.

**Pull request merged**